bears/vcs/CISkip/SkipCIBear.py

```diff
@@ -0,0 +1,44 @@
+
+from coalib.misc.Shell import run_shell_command
+from bears.vcs.CommitBear import _CommitBear
+
+
+class SkipCIBear(_CommitBear):
```

Comment:
an extra level is needed in the abstraction, which needs to be a separate PR before this.

this isnt creating a meta bear.

A meta bear is like `URLBear`, which only yields `HiddenResult`.  See `URLHeadBear` and `InvalidLinkBear` for how they all link together.

We need a `VCSCommitBear` which only yields a `CommitResult(HiddenResult)`.  `SkipCIBear` would then receive those `CommitResult` and only yield the `CommitResult` if the commit was a CI skip.

(and while trying to understand this, it is better to simply ignore `GitCommitBear` and `HgCommitBear` -- those will need to be glued back in after the redesign.)

Reply:
Let me see if I am getting this straight.
I have to make a bear that would return information about the HEAD commit (as HiddenResult) such as whether it is a special commit and possibly a list of modified files. Perhaps an output like:

```
[<CommitResult object(id=0x111dae7389c04f699a0ccc9e4520fccf, 
origin='VCSCommitBear', 
message='HEAD commit information', 
contents=
	({
	  	'commit_type': ['revert_commit', 'ci_skip_commit', ''], 
	  	'modified_files': ['bears/general/InvalidLinkBear.py', ''], 
	  	'commit_sha': 'd83ddce283a58326074b57810dc3aec84e30e8e8'
	 },
	)) 
at 0x7f6f3894dba8>
]
```
And this information can then be used by `SkipCIBear`, `CheckMergeCommitBear` and `CheckRevertCommitBear` respectivey.

Reply:
Also all these bears would be independent of `_CommitBear` so they will be mentioned in the `commit` section of `.coafile` separately.

Reply:
Hey @kriti21 I think what john says makes a lot of sense here. Abstraction is much better and then we can play around with other bear's using it.